Mathlib/MeasureTheory/Integral/Periodic.lean

```diff
@@ -221,14 +226,61 @@ protected theorem intervalIntegral_preimage (t : ℝ) (f : UnitAddCircle → E)
 
 end UnitAddCircle
 
-variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+
+/-!
+## Interval integrability of periodic functions
+-/
 
 namespace Function
 
 namespace Periodic
 
+variable {E : Type*} [NormedAddCommGroup E]
+
 variable {f : ℝ → E} {T : ℝ}
 
+/-- A periodic function is interval integrable over every interval if it is interval integrable
+over one period. -/
+theorem intervalIntegrable {t : ℝ} (h₁f : Function.Periodic f T) (hT : 0 < T)
+  (h₂f : IntervalIntegrable f MeasureTheory.volume t (t + T)) (a₁ a₂ : ℝ) :
+    IntervalIntegrable f MeasureTheory.volume a₁ a₂ := by
+  -- Replace [a₁, a₂] by [t - n₁ * T, t + n₂ * T], where n₁ and n₂ are integers
+  obtain ⟨n₁, hn₁⟩ := exists_nat_ge ((t -min a₁ a₂) / T)
+  obtain ⟨n₂, hn₂⟩ := exists_nat_ge ((max a₁ a₂ - t) / T)
+  have : Set.uIcc a₁ a₂ ⊆ Set.uIcc (t - n₁ * T) (t + n₂ * T) := by
+    apply Set.uIcc_subset_uIcc_iff_le.mpr
+    constructor
+    · calc min (t - ↑n₁ * T) (t + ↑n₂ * T)
+      _ ≤ (t - ↑n₁ * T) := by apply min_le_left
+      _ ≤ min a₁ a₂ := by linarith [(div_le_iff₀ hT).1 hn₁]
+    · calc max a₁ a₂
+      _ ≤ t + n₂ * T := by linarith [(div_le_iff₀ hT).1 hn₂]
+      _ ≤ max (t - ↑n₁ * T) (t + ↑n₂ * T) := by apply le_max_right
+  apply IntervalIntegrable.mono_set _ this
+  clear this
```

Comment:
As below: do you need this line? `this` in particular is being changed all the time
```suggestion

Reply:
The line is not needed for function, only for readability. I followed your suggestion.